strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/providers/layout/Maven2LayoutProvider.java

```diff
@@ -215,27 +197,22 @@ public void deleteFromIndex(RepositoryPath path)
             throws IOException
     {
         Repository repository = path.getFileSystem().getRepository();
-        RepositoryFileAttributes a = Files.readAttributes(path, RepositoryFileAttributes.class);
-
-        if (!repositoryFeatures.isIndexingEnabled(repository) || a.isMetadata())
+        if (!repositoryFeatures.isIndexingEnabled(repository) || isMetadata(path.toString()))
         {
             return;
         }
 
         final RepositoryIndexer indexer = getRepositoryIndexer(path);
         if (indexer != null)
         {
-            String extension = path.getFileName().toString().substring(
-                    path.getFileName().toString().lastIndexOf('.') + 1);
-
-            MavenArtifactCoordinates coordinates = (MavenArtifactCoordinates) a.getCoordinates();
+            Artifact artifact = ArtifactUtils.convertPathToArtifact(path.relativize().toString());
 
             indexer.delete(Collections.singletonList(new ArtifactInfo(repository.getId(),
-                                                                      coordinates.getGroupId(),
-                                                                      coordinates.getArtifactId(),
-                                                                      coordinates.getVersion(),
-                                                                      coordinates.getClassifier(),
-                                                                      extension)));
+                                                                      artifact.getGroupId(),
+                                                                      artifact.getArtifactId(),
+                                                                      artifact.getVersion(),
+                                                                      artifact.getClassifier(),
+                                                                      artifact.getType())));
```

Comment:
I am not entirely sure whose version is more correct. To get better perspective, would you mind telling me why you chose `artifact.getType()` here?
I have some distant memory that version 6.0 of the Maven Indexer was leaning towards dropping the use of `artifact.getType()`... (I may be wrong).

Reply:
> I am not entirely sure whose version is more correct.

If you mean that I dropped previous implementation:
`RepositoryFileAttributes a = Files.readAttributes(path, RepositoryFileAttributes.class);` please note that this solution required physical file existence in the file system. In terms of group repositories, artifacts won't be existing physically.

> To get better perspective, would you mind telling me why you chose artifact.getType() here?

Please notice where did the `Artifact artifact` variable come from. It has been created from `org.carlspring.maven.commons.util.ArtifactUtils#convertPathToArtifact`. Please take a look at the `type` field:
https://github.com/carlspring/maven-commons/blob/master/src/main/java/org/carlspring/maven/commons/util/ArtifactUtils.java#L129

If you compare it with the previous implementation `String extension = path.getFileName().toString().substring(path.getFileName().toString().lastIndexOf('.') + 1);` you'll find out that it's logically the same.